sklearn/utils/extmath.py

```diff
@@ -688,21 +688,41 @@ def _incremental_mean_and_var(X, last_mean=.0, last_variance=None,
     # old = stats until now
     # new = the current increment
     # updated = the aggregated stats
+    if not isinstance(last_sample_count, np.ndarray):
+        # when the index is -1 it will pick the last value in shape array
+        last_sample_count *= np.ones(X.shape[-1])
     last_sum = last_mean * last_sample_count
-    new_sum = X.sum(axis=0)
-
-    new_sample_count = X.shape[0]
+    sum_func = np.nansum if ignore_nan else np.sum
+    new_sum = sum_func(X, axis=0)
+    if not isinstance(new_sum, np.ndarray):
+        new_sum *= np.ones(X.shape[-1])
```

Comment:
We don't need that. X will always be 2D and the sum should always be an ndarray isn't it?